### PR TITLE
MTL-1949: conversion to IMS_IMAGE_ID instead of k8s/ceph version

### PR DESCRIPTION
**Pull request by heemstra**
### Summary and Scope

Pairs with: https://github.com/Cray-HPE/docs-csm/pull/2886

- Fixes: MTL-1949

#### Issue Type

- RFE Pull Request

### Prerequisites

- [ ] I have included documentation in my PR (or it is not required)
- [ ] I tested this on internal system (if yes, please include results or a description of the test)
- [ ] I tested this on a vshasta system (if yes, please include results or a description of the test)

### CODE REVIEW

File: cmd/handoff-bss-metadata.go [hunk 1]
```diff
@@ -213,23 +167,20 @@ func getKernelCommandlineArgs(ncn sls_common.GenericHardware, cmdline string) st
...
+				imsImageID = kubernetesImsImageID
 			}
 
-			if version == "" {
+			if imsImageID == "" {
 				log.Fatalf("ERROR: Could not determine version, version was empty.")
 			}
 
-			cmdlineParts[i] = fmt.Sprintf("metal.server=%s/%s/%s/%s", s3Prefix, path, version, rootfsName)
+			cmdlineParts[i] = fmt.Sprintf("metal.server=%s/%s/rootfs", s3Prefix, imsImageID)
```
Comment (rustydb) [inline cmd/handoff-bss-metadata.go:183]: Why not use `rootfsName` incase it ever ever ever changes again?

**Review by rustydb — CHANGES_REQUESTED** [review]

File: cmd/handoff-bss-metadata.go [hunk 2]
```diff
@@ -101,62 +96,30 @@ var handoffBSSMetadataCmd = &cobra.Command{
...
-		// Validate a given string is a valid version without extra characters prepended or appended.
-		versionRegex, err := regexp.Compile(`(^([0-9a-zA-Z.]+)+(-\d+)?$)`)
-		if err != nil {
-			fmt.Println(err)
-		}
-
-		// Validate the version in a given filename is valid without extra characters prepended or appended.
-		fileVersionRegex, err := regexp.Compile(`(([0-9a-zA-Z.]+)+(-\d+)?).squashfs`)
+		// Validate a given string is a valid UUID
+		versionRegex, err := regexp.Compile(`^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$`)
```
Comment (rustydb) [inline cmd/handoff-bss-metadata.go:100]: Nice.

**Review by rustydb — APPROVED** [review]

File: cmd/handoff-bss-metadata.go (hunk 2 above)
Comment (jacobsalmela) [reply cmd/handoff-bss-metadata.go:100]: Kudos

**Review by jacobsalmela — APPROVED** [review]
👋

File: cmd/handoff-bss-metadata.go (hunk 2 above)
Comment (heemstra) [reply cmd/handoff-bss-metadata.go:100]: Internet FTW

File: cmd/handoff-bss-metadata.go [hunk 3]
```diff
@@ -101,62 +96,30 @@ var handoffBSSMetadataCmd = &cobra.Command{
...
-			if desiredKubernetesVersionMatch == nil {
-				log.Fatalf("ERROR: Could not determine version from [%s]", kubernetesFile)
-			} else {
-				desiredKubernetesVersion = desiredKubernetesVersionMatch[1]
-			}
-
+		// Validate the input is a valid UUID
+		kubernetesImageIDMatch := versionRegex.FindStringSubmatch(kubernetesUUID)
+		if kubernetesImageIDMatch == nil {
+			log.Fatalf("ERROR: Could not determine kubernetes image id from [%s]", kubernetesUUID)
```
Comment (mharding-hpe) [inline cmd/handoff-bss-metadata.go:112]:
```suggestion
			log.Fatalf("ERROR: Could not determine Kubernetes image ID from [%s]", kubernetesUUID)
```

File: cmd/handoff-bss-metadata.go [hunk 4]
```diff
@@ -101,62 +96,30 @@ var handoffBSSMetadataCmd = &cobra.Command{
...
-			if desiredCephVersionMatch == nil {
-				log.Fatalf("ERROR: Could not determine version from [%s]", storageCephFile)
-			} else {
-				desiredCephVersion = desiredCephVersionMatch[1]
-			}
+
+		// Validate the input is a valid UUID
+		cephImsImageIDMatch := versionRegex.FindStringSubmatch(storageUUID)
+		if cephImsImageIDMatch == nil {
+			log.Fatalf("ERROR: Could not determine storage image id from [%s]", storageUUID)
```
Comment (mharding-hpe) [inline cmd/handoff-bss-metadata.go:120]:
```suggestion
			log.Fatalf("ERROR: Could not determine storage image ID from [%s]", storageUUID)
```

File: cmd/handoff-bss-metadata.go [hunk 5]
```diff
@@ -185,22 +148,13 @@ func init() {
...
-	handoffBSSMetadataCmd.Flags().StringVar(&storageCephFile, "storage-ceph-file",
-		"", "The storage-ceph squashFS file to parse version information from (provide this or --storage-ceph-version)")
-	_ = handoffBSSMetadataCmd.MarkFlagFilename("storage-ceph-file")
-
-	handoffBSSMetadataCmd.MarkFlagsRequiredTogether("kubernetes-version", "storage-ceph-version")
-	handoffBSSMetadataCmd.MarkFlagsRequiredTogether("kubernetes-file", "storage-ceph-file")
-	handoffBSSMetadataCmd.MarkFlagsMutuallyExclusive("kubernetes-version", "kubernetes-file")
-	handoffBSSMetadataCmd.MarkFlagsMutuallyExclusive("storage-ceph-version", "storage-ceph-file")
+	handoffBSSMetadataCmd.Flags().StringVar(&kubernetesUUID, "kubernetes-ims-image-id",
+		"", "The kubernetes IMS_IMAGE_ID UUID value")
```
Comment (mharding-hpe) [inline cmd/handoff-bss-metadata.go:152]:
```suggestion
		"", "The Kubernetes IMS_IMAGE_ID UUID value")
```

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**